submodules/timeofday/timeofday.js

```diff
@@ -419,7 +419,7 @@ define(function(require) {
 			if (form_data.start_date === '') {
 				delete form_data.start_date;
 			} else {
-				form_data.start_date = monster.util.dateToEndOfGregorianDay(form_data.start_date, 'UTC');
+				form_data.start_date = monster.util.validateEndOrBeginingOfGregorianDay(form_data.start_date, 'UTC');
```

Comment:
Sorry, I did not check the code thoroughly until yesterday.

It seems that the problem is that the data is loaded in the view using the first time zone available for the following: user timezone, account timezone or local timezone. This is because [it uses the handlebar helper `toFriendlyDate`
](https://github.com/2600hz/monster-ui-callflows/blob/master/submodules/timeofday/views/callflowEdit.html#L105), which internally uses the method [`monster.util#getCurrentTimeZone()`](https://docs.2600hz.com/ui/docs/monster-util/getCurrentTimeZone%28%29/) to determine the timezone to use.

Given this, the best fix would be to use the same method to get the timezone when saving the date, like this:

```suggestion
				form_data.start_date = monster.util.dateToBeginningOfGregorianDay(form_data.start_date, monster.util.getCurrentTimeZone());
```

Also, in that sense, the PRs for `monster-ui` will no longer be necessary.

Reply:
Yes! this solved the issue without modifying on monster-ui, thanks so much for the help! 👍